lib/NativeTestMocks/MockBarometer.cpp

```diff
@@ -0,0 +1,60 @@
+//
+// Created by ramykaddouri on 10/29/24.
+//
+
+#include "MockBarometer.h"
+
+#include <iostream>
+
+MockBarometer::MockBarometer(const std::filesystem::path &dataPath, const std::string& pressureColName, const std::string& temperatureColName) :
+    dataReader(dataPath), pressureColName(pressureColName), temperatureColName(temperatureColName) {
+    Sensor::setName("MockBarometer");
+}
+
+bool MockBarometer::init() {
+    // std::cout << "pressureColName: " << pressureColName << std::endl;
+    // std::cout << "tempColName: " << temperatureColName << std::endl;
+    if(!dataReader.isInit()) return false;
+
+    int numCols = -1;
+    std::string colNames[MAX_NUM_COLS];
+    dataReader.readColumnHeader(numCols, colNames);
+
+    if(numCols == -1 || numCols > MAX_NUM_COLS) {
```

Comment:
Good idea, I'll add it

Reply:
Actually this check is kind of redundant, all of the sensors have an additional check after this to make sure that the indices for the column header strings have been found. If the number of columns read ends up being less than what is expected, the second check will fail since at least one of the column names won't have a corresponding index.